File: little-bee/controller/router/control_command_record.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_CONTROL_COMMAND_RECORD = "/config/control-command-record"
const URL_ROUTER_CONTROL_COMMAND_RECORD_BY_ID = "/config/control-command-record/:control-command-record-id"
const URL_ROUTER_CONTROL_COMMAND_RECORD_QUERY = "/config/control-command-record/query"
const URL_ROUTER_CONTROL_COMMAND_RECORD_ALL = "/config/control-command-record/all"

func ControlCommandRecordRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_CONTROL_COMMAND_RECORD, config.CreateControlCommandRecord)
	r.GET(URL_ROUTER_CONTROL_COMMAND_RECORD_BY_ID, config.RetrieveControlCommandRecord)
	r.PUT(URL_ROUTER_CONTROL_COMMAND_RECORD, config.UpdateControlCommandRecord)
	r.POST(URL_ROUTER_CONTROL_COMMAND_RECORD_QUERY, config.QueryControlCommandRecord)
	r.GET(URL_ROUTER_CONTROL_COMMAND_RECORD_ALL, config.GetAllControlCommandRecords)
	return r
}

File: frontend/console/src/api/control-command-record.js
import Axios from "@/http";

export function createControlCommandRecord(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/control-command-record', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveControlCommandRecord(controlCommandRecordId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/control-command-record/${ controlCommandRecordId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateControlCommandRecord(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/control-command-record', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteControlCommandRecord(controlCommandRecordId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/control-command-record/${ controlCommandRecordId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteControlCommandRecordWithFlag(controlCommandRecordId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/control-command-record/${ controlCommandRecordId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllControlCommandRecords() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/control-command-record/all').then(res => {
            resolve(res)
        })
    })
}

export function queryControlCommandRecord (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/control-command-record/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}